pkg/proxy/integrations/mongoparser/operation.go

```diff
@@ -441,17 +441,25 @@ func decodeOpMsgSectionSequence(section string) (string, string, error) {
 
 func extractSectionSingle(data string) (string, error) {
 	// Look for the prefix before the actual content
-	prefix := "{ SectionSingle msg: "
-	startIndex := strings.Index(data, prefix)
+	prefix1 := "{ SectionSingle msg: "
```

Comment:
There are 2 types of sections for the OpMsg. This method is to extract the `kind: 0`. The error is because of the `kind: 1` is tried to be extracted using this function. There is a decodeOpMsgSectionSequence just above this function which should be used for `kind: 1` documents.